operators/pkg/controller/elasticsearch/driver/default.go

```diff
@@ -568,6 +512,161 @@ func removePodFromList(pods []corev1.Pod, pod corev1.Pod) []corev1.Pod {
 	return pods
 }
 
+func (d *defaultDriver) reconcileNodeSpecs(
+	results *reconciler.Results,
```

Comment:
would a small refactor to not pass this in make sense? I think we intended this to be mostly a return value? (right now both results and the error return value may carry an error).